### ISSUE TITLE
[Bug]: FCM credentials stopped working — previously working credentials now rejected

### ISSUE THREAD

**Issue by Varela949** [bvis/aegis-hass#421]
### Describe the bug

FCM push notifications stopped working today.

The FCM credentials I was using had been working correctly with Aegis for Ajax until today. I have not changed the credentials or my Ajax account configuration.

I therefore extracted the FCM credentials again from the latest Ajax to check whether Ajax had changed the Firebase project or credentials.

The newly extracted credentials are also rejected by Aegis.
Even tried both Auth Keys.

The credentials pass the expected structure checks:

- FCM Project ID: [REDACTED]
- FCM App ID: `1:XXXXXXXXXXXX:android:XXXXXXXXXXXX`
- FCM Sender ID: `XXXXXXXXXXXX`
- FCM API Key: `AIza...` [REDACTED]

The Sender ID matches the numeric part of the App ID.

I have also verified that all four values were extracted from the same current Ajax Android build.

### Steps to reproduce

settings - -  fcm 

### Expected behavior

The FCM credentials extracted from the current Ajax Android app should be accepted and allow the FCM push client to connect.

### Home Assistant version

2026.8.1

### Integration version

1.16.2

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text
2026-08-10 12:14:55.029 WARNING (MainThread) [custom_components.aegis_ajax.notification] FCM credentials were already rejected by Google and haven't changed — skipping re-registration to avoid repeated failed requests against the Firebase project. Re-enter the four values via the Repair card under Settings → Repairs to try again.
```

**Comment by bvis** [comment]
Thank you for the structured report — the shape checks passing and the Sender ID matching the App ID already rule out the most common paste problems.

One important reading first: **the line you pasted is not Google rejecting anything.** It is the integration's own skip-guard, and it fires only when the four values currently in force are byte-identical to a set Google *already* rejected earlier — it deliberately skips the network attempt to avoid hammering the Firebase project with a known-bad set. Two consequences:

- If your re-extraction yielded the same four values as before (expected when the app hasn't changed its project — and current builds carry the same values they have for months), re-entering them shows exactly this message again, without Google ever being asked.
- A genuinely *different* API key does produce a real attempt, and the real attempt logs Google's actual refusal reason. So if both of your Auth Key candidates were truly tried, there are two more informative lines in your log than the one you found.

The story of what actually happened is in the log from the moment push first stopped. Could you search back to that moment and paste the integration's WARNING/ERROR lines from around it? In particular any of these:

1. `Discarded the stored FCM registration and requesting a new one` — if present, this dates the trigger. The four credential values are only used when a **new** registration is created; a working install can run for months on its stored registration, so a wrong or stale value can stay hidden until the registration is discarded. That would explain "worked until today with unchanged credentials" exactly.
2. The first FCM failure line after that — it names Google's actual reason (`API_KEY_INVALID`, `PERMISSION_DENIED`, `ANDROID_APP_BLOCKED`, …). Each of those points to a different fix, which is why the reason matters more than the fact of the rejection.
3. Whether anything happened right before it stopped: Home Assistant upgrade to 2026.8.1, the integration update to 1.16.2, a restart, or re-entering the credentials.

And one check that settles the key question in a single round-trip, without posting any secret: hash each of your API key candidates and give me the hashes plus their character length (a real key is exactly 39 characters):

```bash
printf %s 'AIza…your-key…' | shasum -a 256
```

PowerShell equivalent:

```powershell
$k = 'AIza…your-key…'; $k.Length; [BitConverter]::ToString([Security.Cryptography.SHA256]::Create().ComputeHash([Text.Encoding]::UTF8.GetBytes($k))).Replace('-','').ToLower()
```

For comparison, hashed the same way:

- `65fa2f24…` (starts with) — this is a value that greps easily out of the app but is **not** an FCM key (it belongs to a different Google service); if one of yours matches this, that is the whole story.
- `ca34fb16…` and `31a4a0a2…` — the two known-good FCM keys of the official app's two Firebase projects. If one of yours matches either of these, the key is right and the problem is elsewhere (likely the pairing with the other three values, or the original rejection reason from point 2).

With the first-failure lines and the two hashes, this should resolve in one more round.

**Comment by Varela949** [comment]
after some hours it accepted the fcm credentials.
sorry for that

**Comment by bvis** [comment]
No apology needed — glad it recovered. "Accepted after some hours" with unchanged values reads like a transient refusal on Google's side, which is genuinely useful to know about. If it ever happens again, the two things worth grabbing at that moment are the first FCM WARNING lines from when push stops (they name Google's actual reason) — with those, a transient like this is distinguishable from a real credential problem in one look.
